clang/lib/Sema/SemaStmt.cpp

```diff
@@ -717,6 +717,12 @@ bool Sema::checkMustTailAttr(const Stmt *St, const Attr &MTA) {
     return false;
   }
 
+  if (const FunctionDecl *CalleeDecl = CE->getDirectCallee();
+      CalleeDecl && CalleeDecl->hasAttr<NotTailCalledAttr>()) {
+    Diag(St->getBeginLoc(), diag::err_musttail_mismatch) << true << CalleeDecl;
+    return false;
```

Comment:
Strange here:  We typically do a 'return true' in check functions on failure, so surprised this is a return false, but i see everywhere here does it.

Reply:
Yep, exactly :/  Commented mostly so that someone someday might come along and do the work to figure out what is going on here :)